src/docs/reference/README.md

```diff
@@ -0,0 +1,33 @@
+# Reference of Built-in Modules
+
+This is a comprehensive reference for the modules and their features available in Orchard Core. Under each of these pages, you'll find an overview of the given module with documentation on how to use and extend them.
+
+Check out the reference pages for each module under the following categories in the menu:
+
+- CMS Modules: Modules that contain content management features that all interact with content items in some way. E.g. the base content management module, the media management module, or modules related to search and indexing are here.
+- Core Modules: Modules that usually provide some fundamental features for a web app which is not connected to content management, or framework-like functionality that you can build on. These modules include e.g. user and role management, or the UI localization and background task management infrastructure.
+
+The above distinction is also present in the Orchard Core source code with solution folders. However, this is just a logical categorization and technically all the modules are alike.
+
+## Overview of all Orchard Core features
+
+
+### Admin
+
+- [Admin (`OrchardCore.Admin`)](modules/Admin/README.md)
+- [Admin Menu (`OrchardCore.AdminMenu`)](modules/AdminMenu/README.md)
+- [Custom Settings (`OrchardCore.CustomSettings`)](modules/CustomSettings/README.md)
+
+### Content
+
+### Core
+
+### Design
+
+### Extensibility
+
+### Localization
+
+### Search
+
+### Security
```

Comment:
I started to do this to achieve something similar as under https://github.com/OrchardCMS/OrchardCore/pull/6942 but I'm not sure it's worth it, and maintaining it. Maybe we should categorize features in submenus instead, not just have a Core-CMS (and the couple of other lower levels) distinction?

Reply:
If we decide to list the features per section, we have to list them all (A Task list like https://facelessuser.github.io/pymdown-extensions/extensions/tasklist/ could be useful to show what is implemented and what is planned), but I agree that it is complicated to maintain.

We could just keep the sections without features/links and add a little description for each one.

Reply:
Absolutely, as said I'm just not sure about continuing this. Wouldn't a multi-level menu with a different structure than Core/CMS modules be better instead of a list here?

Reply:
This is a Reference doc section so I guess that the decision to use a structure that is close to the solution folders was good (hence the titles with the module namespace).
I don't mind if we organize them otherwise.

Reply:
I've reordered and categorized the pages in the menu. What do you think?

Probably the issue with this is that while it gives structure and clarity it reduces discoverability: You're less likely to stumble upon something you actually need but don't know you're looking for it, by chance, since it's buried under a submenu (especially that the categories are kind of opinionated). So maybe having these categories but in a flat list on the page is better (I'd use this categorization though, slightly different to what you've previously created).

Reply:
@agriffard I had a question for you here. This is not done yet because it's a question what kind of categorization we want: submenus or a flat list on a page. I'm not sure about either one but have completed the menu and can complete the list; please check out how it looks for you.

Reply:
As I said, we can organize it we the sub menus structure you suggested, but it is indeed better if the reference root page can give a good overview of the pages in each section.

Reply:
Can you check out the menu specifically, please, and tell how well it works or not for you?

Reply:
See the recorded screen to gif.

Reply:
I mean, I know how it looks like :). I'm interested in how do you feel when looking at the docs site, is the multi-level menu more useful, or the flat list?

Reply:
The advantage of sticking with the folder structure is that it is easy to know where to add a new doc link in the TOC.

but I think that people are mainly using the search when they want to find a specific doc.
That's why I don't mind.

Reply:
I've restored the menu and added a categorized list here (and fixed a few issues I've found). What do you think?